### PR TITLE
Remove unwrapping of OCTET STRING wrapping in EnvelopedCms.

### LINKED ISSUE #68554: EnvelopedCms Decryption of binary data corrupts response on Linux if unencrypted data starts with byte value 0x04
### Description

Encrypting a data buffer that begins with byte value 0x04 fails to properly decrypt on Linux.
The same encryption/decryption operation works properly on Windows.

### Reproduction Steps

Run the attached Visual Studio project to demonstrate failure
[CheckEncryption.zip](https://github.com/dotnet/runtime/files/8563915/CheckEncryption.zip)
.

### Expected behavior

Encryption of the unencrypted buffer followed by decryption of the encrypted buffer should produced the original unencrypted buffer.

### Actual behavior

If the unencrypted buffer starts with byte value 0x04, the decryption of the encrypted buffer produces a buffer that is different from the unencrypted buffer.

### Regression?

This test fails for Linux Centos7 for using release versions of netcoreapp3.1 linux-x64, net5.0 linux-x64 and net6.0 linux-x64.
It has not been tested on other versions of Linux.
This test does not fail on Windows.

### Known Workarounds

Our workaround is to prepend pad bytes before the data to be encrypted, and then to strip these out after decryption to insure that the unencrypted buffer does not start with a leading EOF (0x04) byte.

### Configuration

DotNet version: 6.0.100 targeting net6.0, net5.0 or netcoreapp3.1
OS and version: CentOS Linux release 7.9.2009 (Core)
Architecture: linux-x64
Problem exists on this Linux version; does not occur on Windows; has not been tested on other Linux distros.

### Other information

_No response_

### PR DESCRIPTION
**Pull request by vcsjones**
Prior to .NET Core 3.0, EnvelopedCms would pack and unpack PKCS7 data in an
OCTET STRING when encrypted and decrypted. This turned out to be incorrect.
Other implementations do not do the OCTET STRING wrapping, so when a different
CMS implementation processed a PKCS7 data encrypted by .NET, it would contain
the raw OCTET STRING. So the data would appear to be appended by the byte 0x04,
and then a length indicator.

We removed the wrapping in OCTET STRING. However, we left in the unwrapping of
the OCTET STRING because previously encrypted data by .NET would now contain
the raw data, thus appearing prefixed by 0x04 and a length indicator.

This however still has a problem where if the data to be encrypted started with
0x04 and a reasonable length, it would mistakenly be treated as an OCTET STRING.

This change removes the compatibility and is thus a breaking change. If customers
have CMS data that was encrypted by .NET Core prior to 3.0, they will experience
decrypted data that starts with the byte 0x04 and at least one length byte.

Customers can work around this with AsnDecoder.ReadOctetString to remove the extra
bytes, and optionally re-encrypting it and storing that so the work around is no
longer necessary.

Closes #68554.

### CODE REVIEW

**Comment by vcsjones** [comment]
Not marking as `breaking-change` until we're sure we want to take the change.

File: src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/GeneralTests.cs [hunk 1]
```diff
@@ -378,37 +378,6 @@ public static void Encrypt_Data_DoesNotIncreaseInSize()
...
-            {
-                if (privateCert == null)
-                {
-                    return; //Private key not available.
-                }
-
-                ecms.Decrypt(new X509Certificate2Collection(privateCert));
-            }
-
-            Assert.Equal(expectedContent, ecms.ContentInfo.Content);
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/GeneralTests.cs:409]: Would it be worthwhile to keep this test, make it available to all platforms, and show that we're returning the message as-is (aka that we've definitely deleted the compat code?)

Comment (vcsjones) [reply src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/GeneralTests.cs]: Hm, yeah that makes sense, so, done.

**Comment by vcsjones** [comment]
Outerloop failures appear unrelated.

**Review by bartonjs — APPROVED** [review]

**Comment by bartonjs** [comment]
Created the breaking change documentation: https://github.com/dotnet/docs/issues/29263

**Pull request merged**